client/components/token-field/index.jsx

```diff
@@ -51,37 +49,33 @@ const TokenField = createReactClass( {
 				}
 			} );
 		},
-	},
-
-	getDefaultProps: function() {
-		return {
-			suggestions: Object.freeze( [] ),
-			maxSuggestions: 100,
-			value: Object.freeze( [] ),
-			placeholder: '',
-			displayTransform: identity,
-			saveTransform: function( token ) {
-				return token.trim();
-			},
-			onChange: function() {},
-			isBorderless: false,
-			disabled: false,
-			tokenizeOnSpace: false,
-		};
-	},
-
-	mixins: [ PureRenderMixin ],
-
-	getInitialState: function() {
-		return {
-			incompleteTokenValue: '',
-			inputOffsetFromEnd: 0,
-			isActive: false,
-			selectedSuggestionIndex: -1,
-			selectedSuggestionScroll: false,
-			tokenInputHasFocus: false,
-		};
-	},
+	};
+
+	static defaultProps = {
+		suggestions: Object.freeze( [] ),
+		maxSuggestions: 100,
+		value: Object.freeze( [] ),
+		placeholder: '',
+		displayTransform: identity,
+		saveTransform: function( token ) {
+			return token.trim();
+		},
+		onChange: function() {},
+		isBorderless: false,
+		disabled: false,
+		tokenizeOnSpace: false,
+	};
+
+	static initialState = {
```

Comment:
Just a warning, I've been burned by doing this where `initialState` was mutated in non-obvious ways which leads to a mess. Fortunately, that doesn't seem to be a danger here.

We could add a bit of safety in one of the following ways:
* Slightly safer with limited added overhead: `Object.freeze` this property.
* Totally safe, more overhead: use a function that returns a new object.

Either of these would add a bit of safety but aren't necessary.

Reply:
Yeah, I was trying to assess the risk here, too, but since we're only mutating state via `setState()`, I think we should be fine.

> Just a warning, I've been burned by doing this where `initialState` was mutated in non-obvious ways which leads to a mess.

Can you give me an example? I was wondering if any such occurrence was maybe due to use of some anti-pattern...

Reply:
This is fine to ship now as it seems quite safe. Any safety measures might serve as a sort of warning to future devs. I'll see about contriving an example of bad state mutation…